### PR TITLE
Update SPU Default Ports

### PR DESCRIPTION
**Pull request by AnthonyMBonafide**
Change default SPU public port to 9010

Update local cluster to use the project wide default SPU public port

Fixes https://github.com/infinyon/fluvio/issues/3739

### CODE REVIEW

**Comment by EstebanBorai** [comment]
> Change default SPU public port to 9010
> 
> Update local cluster to use the project wide default SPU public port
> 
> Fixes #3739

Thanks so much @AnthonyMBonafide!

File: crates/fluvio-cluster/src/runtime/local/spu.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@ use std::process::{Command, Stdio};
 use std::{fs::File, path::PathBuf};
 
 use anyhow::{Result as AnyResult, anyhow};
+use fluvio_types::defaults::SPU_PUBLIC_PORT;
 use tracing::{debug, info, instrument};
 
 use fluvio_controlplane_metadata::spu::{Endpoint, IngressAddr, IngressPort, SpuSpec, SpuType};
```
Comment (EstebanBorai) [inline crates/fluvio-cluster/src/runtime/local/spu.rs:8]:
```suggestion
use tracing::{debug, info, instrument};

use fluvio_types::defaults::SPU_PUBLIC_PORT;
use fluvio_controlplane_metadata::spu::{Endpoint, IngressAddr, IngressPort, SpuSpec, SpuType};
```

Comment (EstebanBorai) [reply crates/fluvio-cluster/src/runtime/local/spu.rs:8]: This is nitpick, but we try to group together local workspace imports!

Comment (sehz) [reply crates/fluvio-cluster/src/runtime/local/spu.rs:8]: Yes, Fluvio standard is 
- std lib
- third party
- fluvio crates

**Comment by sehz** [comment]
Other than import ordering, looks good

**Comment by AnthonyMBonafide** [comment]
Excellent. Thank you for the feedback. I'll get this fixed up a little later.

**Comment by AnthonyMBonafide** [comment]
I accepted the suggested changes, thank you @EstebanBorai  for that. Should I squash and rebase or will that be done via the Github UI during the merge process?

**Comment by EstebanBorai** [comment]
> I accepted the suggested changes, thank you @EstebanBorai  for that. Should I squash and rebase or will that be done via the Github UI during the merge process?

Thanks to you @AnthonyMBonafide!

No need to squash!

**Review by EstebanBorai — APPROVED** [review]

**Comment by EstebanBorai** [comment]
@AnthonyMBonafide can you please update the patch number in the Fluvio Types crate?
https://github.com/infinyon/fluvio/actions/runs/7523340311/job/20491819953?pr=3815#step:13:244

**Comment by sehz** [comment]
9005 is still referencing in the helm chart: https://github.com/infinyon/fluvio/blob/master/k8-util/helm/fluvio-app/templates/sc-deployment.yaml

File: crates/fluvio-cluster/src/runtime/local/sc.rs [hunk 2]
```diff
@@ -54,7 +54,7 @@ impl ScProcess {
         };
 
         if let TlsPolicy::Verified(tls) = &self.tls_policy {
-            self.set_server_tls(&mut binary, tls, 9005)?;
+            self.set_server_tls(&mut binary, tls, 9010)?;
```
Comment (sehz) [inline crates/fluvio-cluster/src/runtime/local/sc.rs:57]: can you change to use constant?

**Review by sehz — APPROVED** [review]
LGTM.  Nice job

**Comment by AnthonyMBonafide** [comment]
> LGTM. Nice job

Thank you. I am excited to contribute more. Thank you for your help. I hope the CI jobs pass though :)

**Comment by EstebanBorai** [comment]
@sehz @AnthonyMBonafide looks like we are missing another definition for SPUs perhaps?

**Comment by AnthonyMBonafide** [comment]
> @sehz @AnthonyMBonafide looks like we are missing another definition for SPUs perhaps?

Could you elaborate on the issue and point me in the right direction? I dug around the code base, I am a complete noob so I apologize, and I couldn't find anything obvious that needed to be adjusted.

**Comment by sehz** [comment]
Hi @AnthonyMBonafide.  Thanks for trying to fix.  troubleshooting shooting CI is not simple and pretty time consuming.  So better way to diagnose is run locally.  Set up fluvio cluster locally, perform upgrade and check log.

**Comment by AnthonyMBonafide** [comment]
Just an update: I am still working on fixing this, I am just going to need some time as I work on this in my spare time and I am taking the extra time to debug through the code.

**Comment by EstebanBorai** [comment]
> Just an update: I am still working on fixing this, I am just going to need some time as I work on this in my spare time and I am taking the extra time to debug through the code. I am looking to understand the codebase better along the way.

Please ask any questions! We are happy to help!

**Comment by AnthonyMBonafide** [comment]
@EstebanBorai , could you point me in the right direction? I am looking for the place where the SC is connecting to the SPU to update its state and continue processing after the SPU is up and running?

**Comment by AnthonyMBonafide** [comment]
I am going to close this out. I am not able to figure this out effectively. I think I will take a stab at another issue.

**Comment by ajhunyady** [comment]
@AnthonyMBonafide, I've reopened this issue, and we'll assign an engineer to work with you and get to the bottom of it.
I'm sorry it took so long; we have been swamped with other issues and did not get to this in time.

**Comment by AnthonyMBonafide** [comment]
> @AnthonyMBonafide, I've reopened this issue, and we'll assign an engineer to work with you and get to the bottom of it. I'm sorry it took so long; we have been swamped with other issues and did not get to this in time.

Thank you for this. Do not apologize, I completely understand, I did not want to keep this going on and start to eat up anyone else's time. I am newer to Rust and don't want to drag anyone down.

**Comment by digikata** [comment]
This was tricky.  The problem being exposed in the CI tests is that the ip addresses consumed in various configurations is not clearly delineated/calculated in the code for tls mode.  This causes a port collision in tls configuration.

The upstream codebase port settings 9005 base, 9010 cluster, work because the setup statically does not cross over port bindings in tls config.  There is no real calculation in the code, the "magic" constants for the ports take care of not overlapping.

This can be seen in the process run tree for the TLS configured CI case that starts the cluster with:
```
AUTH_POLICY=crates/fluvio-sc/test-data/auth_config/policy.json X509_AUTH_SCOPES=crates/fluvio-sc/test-data/auth_config/scopes.json fluvio cluster start --local --develop --spu 2 --rust-log fluvio=info --namespace ${K8_NAMESPACE} --tls --domain fluvio.local --server-key ./tls/certs/server.key --server-cert ./tls/certs/server.crt --ca-cert ./tls/certs/ca.crt --client-cert ./tls/certs/client-root.crt --client-key ./tls/certs/client-root.key
```

The processes started with **stable** fluvio 0.11.4 look like this: (note the `--bind-non-tls-public` address vs the spu 5001 public address:
```
$ ps ax | grep fluvio-run
fluvio-run sc --local .fluvio/data/metadata --tls --enable-client-cert --server-cert ./tls/certs/server.crt --server-key ./tls/certs/server.key --ca-cert ./tls/certs/ca.crt --bind-non-tls-public 0.0.0.0:9005
fluvio-run spu --tls --enable-client-cert --server-cert ./tls/certs/server.crt --server-key ./tls/certs/server.key --ca-cert ./tls/certs/ca.crt --bind-non-tls-public 0.0.0.0:9022 -i 5002 -p 0.0.0.0:9020 -v 0.0.0.0:9021 --log-base-dir /Users/alan/.fluvio/data
fluvio-run spu --tls --enable-client-cert --server-cert ./tls/certs/server.crt --server-key ./tls/certs/server.key --ca-cert ./tls/certs/ca.crt --bind-non-tls-public 0.0.0.0:9012 -i 5001 -p 0.0.0.0:9010 -v 0.0.0.0:9011 --log-base-dir /Users/alan/.fluvio/data
```

With the public port constant realigned, starting fluvio cluster with this PR, spu 5001 is missing. It could not bind 5010, because of the `bind-non-tls-public` setting.

```
$ ps ax | grep fluvio-run
fluvio-run sc --local .fluvio/data/metadata --tls --enable-client-cert --server-cert ./tls/certs/server.crt --server-key ./tls/certs/server.key --ca-cert ./tls/certs/ca.crt --bind-non-tls-public 0.0.0.0:9010
fluvio-run spu --tls --enable-client-cert --server-cert ./tls/certs/server.crt --server-key ./tls/certs/server.key --ca-cert ./tls/certs/ca.crt --bind-non-tls-public 0.0.0.0:9022 -i 5002 -p 0.0.0.0:9020 -v 0.0.0.0:9021 --log-base-dir /Users/alan/.fluvio/data
```

With non tls test configs, this isn't a problem.

**Comment by AnthonyMBonafide** [comment]
@digikata Thank you so much, I will take a look some time tomorrow or this weekend.

**Comment by AnthonyMBonafide** [comment]
@digikata Thank you for all that detail. I am sorry, but I dug around and it still is not popping out to me where the issue is, could you point me to some places in the code where I should look?

**Pull request closed without merging**